ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -447,9 +451,12 @@ generateGhcLibCabal ghcFlavor = do
         ,"data-files:"] ++
         indent dataFiles ++
         ["extra-source-files:"] ++
-        -- Remove Config.hs, Version.hs, Parser.hs and Lexer.hs from
-        -- the list of extra source files here.
-        indent (reverse (drop 4 $ reverse $ extraFiles ghcFlavor)) ++ indent cHeaders ++
+         ( let numFiles = if ghcFlavor == GhcMaster then 4 else 3 in
+             -- Remove Config.hs, Version.hs, Parser.hs and Lexer.hs from
+             -- the list of extra source files here.
+             indent (reverse (drop numFiles $ reverse $ extraFiles ghcFlavor)) ++ indent (cHeaders ghcFlavor)
```

Comment:
This code is trying to be clever and reuse `extraFiles` but I think it actually ends up being harder to read and more fragile than just listing the files that you want to remove here or alternatively factor it into a separate list `generatedFiles ghcFlavor` and turn `extraFiles` into something like `extraFiles ghcFlavor = … ++ generatedFiles ghcFlavor`. I also don’t understand why we remove those files for `ghc-lib-parser.cabal` but keep them for `ghc-lib.cabal`.

Reply:
Isn't it the case that they go into `ghc-lib-parser.cabal` but NOT `ghc-lib.cabal`? Checking... yes.

I take your point about reusing `extraFiles`. It's an artifact of evolution. It certainly can be improved.